xarray/indexes/range_index.py

```diff
@@ -0,0 +1,366 @@
+import math
+from collections.abc import Hashable, Mapping
+from typing import Any
+
+import numpy as np
+import pandas as pd
+
+from xarray.core.coordinate_transform import CoordinateTransform
+from xarray.core.dataarray import DataArray
+from xarray.core.indexes import CoordinateTransformIndex, Index, PandasIndex
+from xarray.core.indexing import IndexSelResult, normalize_slice
+from xarray.core.variable import Variable
+
+
+class RangeCoordinateTransform(CoordinateTransform):
+    """1-dimensional coordinate transform representing a simple bounded interval
+    with evenly spaced, floating-point values.
+    """
+
+    start: float
+    stop: float
+
+    __slots__ = ("start", "stop")
+
+    def __init__(
+        self,
+        start: float,
+        stop: float,
+        size: int,
```

Comment:
don't slow down merging, but would it be safer to make this kwargs-only from here, since `size` is not the standard `step` item? 

(I'm joining late and so I'm guessing there are good reasons to use `size` rather than `step` here...)